src/Symfony/Bundle/DebugBundle/Resources/config/services.xml

```diff
@@ -20,19 +22,19 @@
             <argument>null</argument><!-- var_dumper.cli_dumper when debug.dump_destination is set -->
         </service>
 
-        <service id="debug.dump_listener" class="Symfony\Component\HttpKernel\EventListener\DumpListener">
+        <service id="debug.dump_listener" class="Symfony\Component\HttpKernel\EventListener\DumpListener" public="true">
```

Comment:
100% agree, and this PR is on that path :)
No need for any new feature as there is already a trick to do the deprecation, see:
https://github.com/symfony/symfony/blob/f2f9aa61684d867df280473d56c928e6906aa437/src/Symfony/Bundle/FrameworkBundle/Resources/config/form.xml#L192
So, we should just turn these services as private, then register them in a public dummy registry, that will prevent inlining and will thus make them "public" (but in a deprecated manner).